src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/LocalTypeSpecSdkGenerationService.cs

```diff
@@ -0,0 +1,286 @@
+using Microsoft.Extensions.Logging;
+using Azure.Tools.GeneratorAgent.Configuration;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Tools.GeneratorAgent
+{
+    internal class LocalTypeSpecSdkGenerationService : ISdkGenerationService
+    {
+        private readonly AppSettings AppSettings;
+        private readonly ILogger<LocalTypeSpecSdkGenerationService> Logger;
+        private readonly ProcessExecutor ProcessExecutor;
+        private readonly string TypeSpecSourcePath;
+        private readonly string SdkOutputDirectory;
+
+
+        public LocalTypeSpecSdkGenerationService(
+            AppSettings appSettings,
+            ILogger<LocalTypeSpecSdkGenerationService> logger,
+            ProcessExecutor processExecutor,
+            string typeSpecSourcePath,
+            string sdkOutputDirectory)
+        {
+            AppSettings = appSettings ?? throw new ArgumentNullException(nameof(appSettings));
+            Logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            ProcessExecutor = processExecutor ?? throw new ArgumentNullException(nameof(processExecutor));
+            TypeSpecSourcePath = typeSpecSourcePath ?? throw new ArgumentNullException(nameof(typeSpecSourcePath));
+            SdkOutputDirectory = sdkOutputDirectory ?? throw new ArgumentNullException(nameof(sdkOutputDirectory));
+            if (string.IsNullOrWhiteSpace(TypeSpecSourcePath))
+            {
+                throw new ArgumentException("TypeSpec source path cannot be null or whitespace", nameof(typeSpecSourcePath));
+            }
+
+            if (string.IsNullOrWhiteSpace(SdkOutputDirectory))
+            {
+                throw new ArgumentException("SDK output directory cannot be null or whitespace", nameof(sdkOutputDirectory));
+            }
+        }
+
+        protected virtual bool DirectoryExists(string path) => Directory.Exists(path);
```

Comment:
You can define these common things as a default implementation in the interface or shift to a base class so that you're not writing the same thing over and over.

It may even make sense to extract a "FilerOperations" class for this.   (there are test shim packages that do a nice job here, but none are on our approved list, sadly.)